src/GazeboThruster.cpp

```diff
@@ -175,7 +192,25 @@ void GazeboThruster::updateBegin(common::UpdateInfo const& info)
             thruster != thrusters.end(); ++thruster)
     {
         physics::LinkPtr link = model->GetLink( thruster->name );
-        link->AddRelativeForce( math::Vector3(thruster->effort,0,0) );
+        link->AddLinkForce( thruster->effort*gazebo::math::Vector3::UnitX);
+        link->AddLinkForce( thruster->effort*thruster->direction, thruster->position );
     }
 }
 
+void GazeboThruster::updateCompensatedEffort(gazebo_underwater::Matrix6 const& matrix, gazebo::math::Vector3 const& cog)
+{
+    gazebo_underwater::Vector6 effort;
```

Comment:
>  Just one variable to be used for all thursters

Not entirely true. It's one variable, but it is basically reset for each thruster. A.k.a. it is a different *value* for each thruster.

Defining a variable outside the loop suggests that the value of the variable is carried between one loop to the next. Since it is not the case, make it unambiguous by defining the variable *where it is first used*. Or, more specifically, in the most specific scope possible.